lib/algora/library.ex

```diff
@@ -433,44 +435,46 @@ defmodule Algora.Library do
     Phoenix.PubSub.unsubscribe(@pubsub, topic(channel.user_id))
   end
 
-  defp create_thumbnail_from_file(%Video{} = video, src_path, opts) do
-    dst_path = Path.join(System.tmp_dir!(), "#{video.uuid}.jpeg")
+  defp create_thumbnail_from_file(%Video{} = video, src_path, marker, opts \\ []) do
+    dst_path = Path.join(System.tmp_dir!(), "#{video.uuid}-#{marker.minutes}.jpeg")
 
-    with :ok <- Thumbnex.create_thumbnail(src_path, dst_path, opts) do
-      File.read(dst_path)
+    if not(File.exists?(dst_path)) do
+      :ok = Thumbnex.create_thumbnail(src_path, dst_path, opts)
```

Comment:
Avoids trying to re-create thumbnails that already exist on disk. This is because certain segment numbers hit `Algora.Pipeline.Storage#store_content` more than once (from memory I think it was segment 0)